### PR TITLE
refactor(vendor): unvendor six

### PR DESCRIPTION
**Pull request by jd**
This replaces the vendored six library by a dependency on upstream six 1.7.0,
which is the first version to add `six.wraps` that is being used by at least
the vendored version of debtcollector.

### CODE REVIEW

**Review by brettlangdon — DISMISSED** [review]

**Review by P403n1x87 — DISMISSED** [review]

**Review by Yun-Kim — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by Yun-Kim — APPROVED** [review]

File: releasenotes/notes/depends-on-six-e9b79edbf0d169a2.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+other:
+  - |
+    The `six` library has been removed from `vendor` and the system-wide version is being used.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/depends-on-six-e9b79edbf0d169a2.yaml:4]: Might be worth mentioning the minimum version required as that could be a breaking change is someone has a really old version of six pinned.

File: ddtrace/internal/uds.py [hunk 2]
```diff
@@ -4,17 +4,19 @@
...
 
-class UDSHTTPConnection(httplib.HTTPConnection):  # type: ignore[name-defined]
+class UDSHTTPConnection(httplib.HTTPConnection):
     """An HTTP connection established over a Unix Domain Socket."""
 
     # It's "important" to keep the hostname and port arguments here; while there are not used by the connection
     # mechanism, they are actually used as HTTP headers such as `Host`.
-    def __init__(self, path, https, *args, **kwargs):
-        # type: (str, bool, *Any, **Any) -> None
-        if https:
```
Comment (Kyle-Verhoog) [inline ddtrace/internal/uds.py:14]: idek if it's possible to have an https connection over unix socket or if it's even something we want to support (we definitely don't test it) but should this change be made subtly in this PR?

File: ddtrace/internal/dogstatsd.py [hunk 3]
```diff
@@ -1,12 +1,15 @@
-from typing import Dict
```
Comment (Kyle-Verhoog) [inline ddtrace/internal/dogstatsd.py:1]: These changes seem unrelated to unvendoring six

File: ddtrace/tracer.py [hunk 4]
```diff
@@ -315,7 +315,8 @@ def configure(
         elif writer is None and isinstance(self.writer, LogWriter):
             # No need to do anything for the LogWriter.
             pass
-        self.writer.dogstatsd = get_dogstatsd_client(self._dogstatsd_url)
+        if isinstance(self.writer, AgentWriter):
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:318]: Also seems like an unrelated change

File: ddtrace/sampler.py [hunk 5]
```diff
@@ -35,13 +36,13 @@
 KNUTH_FACTOR = 1111111111111111111
 
 
-class BaseSampler(six.with_metaclass(abc.ABCMeta)):  # type: ignore[misc]
```
Comment (Kyle-Verhoog) [inline ddtrace/sampler.py:38]: 🥳

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]

File: ddtrace/internal/dogstatsd.py (hunk 3 above)
Comment (jd) [reply ddtrace/internal/dogstatsd.py:1]: You wish… they are.

Unvendoring six raises a new bunch of mypy errors. In this particular case, there are a lot of typing information that is lost by moving from a URL `str` to a `Dict` where values have multiple types. That's why I removed this parsing function and merged it, so we don't lose typing information.

File: ddtrace/internal/uds.py (hunk 2 above)
Comment (jd) [reply ddtrace/internal/uds.py:14]: In theory, I think it's possible.
In practice we don't support it since we don't support something like `unix+ssl://` as a scheme.

File: ddtrace/tracer.py (hunk 4 above)
Comment (jd) [reply ddtrace/tracer.py:318]: Another mypy caught bug.
Not all writer have `dogstatsd` as an attribute.

File: ddtrace/internal/dogstatsd.py (hunk 3 above)
Comment (Kyle-Verhoog) [reply ddtrace/internal/dogstatsd.py:1]: Gotcha! I think it makes sense to do the change separately though

Comment (jd) [reply ddtrace/internal/dogstatsd.py:1]: Sure, done in https://github.com/DataDog/dd-trace-py/pull/2289

**Review by Yun-Kim — APPROVED** [review]

**Review by P403n1x87 — APPROVED** [review]
